apps/mochi-web/components/app/NewAppForm.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, Input, Typography } from '@consolelabs/core'
+import {
+  DtoCreateApplicationRequest,
+  ViewFullApplicationResponse,
+} from '~types/mochi-pay-schema'
+import { API } from '~constants/api'
+import { useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+
+type Props = {
+  id?: string
+  onClose: () => void
+  onSuccess?: (result: ViewFullApplicationResponse) => void
+  onError?: () => void
+}
+
+const schema = z.object({
+  app_name: z
+    .string()
+    .min(6, 'Name must contain at least 6 characters')
+    .regex(
+      /^[a-zA-Z0-9 ]+$/,
+      'Name must contain only letters, numbers and spaces',
+    ),
+})
+
+export default function NewAppForm({ id, onClose, onSuccess, onError }: Props) {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm<DtoCreateApplicationRequest>({
+    defaultValues: {
+      app_name: '',
+    },
+    resolver: zodResolver(schema),
+  })
+
+  const onCreateApp = (data: DtoCreateApplicationRequest) => {
```

Comment:
Can we have submitting state and reflect the state on the submit button? (the submit button needs to be disabled then)